src/Plugins/Loader/ManagedLoadContext.cs

```diff
@@ -175,7 +181,9 @@ protected override IntPtr LoadUnmanagedDll(string unmanagedDllName)
             var resolvedPath = _dependencyResolver.ResolveUnmanagedDllToPath(unmanagedDllName);
             if (!string.IsNullOrEmpty(resolvedPath) && File.Exists(resolvedPath))
             {
-                return LoadUnmanagedDllFromPath(resolvedPath);
+                return _loadInMemory
+                    ? LoadUnmanagedDllFromShadowCopy(resolvedPath)
+                    : LoadUnmanagedDllFromPath(resolvedPath);
```

Comment:
`_loadInMemory` does sound a little strange in this case, but I didn't want to rename anything.

Reply:
Thanks for your consideration. For internal-only variables, I'm 100% okay renaming so something that is better.

Reply:
`_hotReloadEnabled` could cover both loading into memory and shadow copying. But then we will end up with something like

`new ManagedLoadContext(... , hotReloadEnabled: _loadInMemory)`

in `AssemblyLoadContextBuilder`, or if the name of the field in ALCB is changed as well

```csharp
public AssemblyLoadContextBuilder PreloadAssembliesIntoMemory()
{
    _hotReloadEnabled = true; // required to prevent dotnet from locking loaded files
    return this;
}
```

Any preferences?

Reply:
How about we separate these behaviors like this? It should just work for users calling `PluginLoader`, but anyone using `AssemblyLoadContextBuilder` now has more granular control over this.

```diff
public ManagedLoadContext(
    // ...
    bool loadInMemory,
+   bool shadowCopyNativeLibraries) 
```
I'm not sure if anyone is using AssemblyLoadContextBuilder directly, but in case they are:
```diff
class AssemblyLoadContextBuilder
{
+     public AssemblyLoadContextBuilder ShadowCopyNativeLibraries()
+      {
+            _shadowCopyNativeLibraries = true;
+           return this;
+      }
}
```
And add `builder.ShadowCopyNativeLibraries()` to https://github.com/natemcmaster/DotNetCorePlugins/blob/370d3558e675e195728d02df5c9bf2f59d258a51/src/Plugins/PluginLoader.cs#L363-L366

Reply:
Sounds good! Is there something special I need to do/keep in mind when adding to the public API?

Reply:
You'll need to add the API to PublicApi.Unshipped.txt. If you're using VS, when you write the new code you'll get red squiggly underlines on the new API. There should be a Quick Action on this to generate the content in PublicApi.Unshipped.txt for you.